src/features/maintenance/useMaintenance.tsx

```diff
@@ -0,0 +1,25 @@
+import { useEffect } from 'react'
+
+import { navigationRef } from 'features/navigation/navigationRef'
+import { remoteStore } from 'libs/remoteStore'
+
+const resetToMaintenancePage = () => {
+  navigationRef.current?.reset({
+    index: 0,
+    routes: [
+      {
+        name: 'Maintenance',
+      },
+    ],
+  })
+}
+
+const onMaintenanceStatusChange = (maintenanceStatus: boolean) => {
+  if (maintenanceStatus) resetToMaintenancePage()
+}
+
+export const useBlockForMaintenance = () => {
+  useEffect(() => {
+    remoteStore.maintenanceStatus(onMaintenanceStatusChange)
```

Comment:
```suggestion
    const subscriber = remoteStore.maintenanceStatus(onMaintenanceStatusChange)
    return () => subscriber()
```

Comme ici: https://rnfirebase.io/firestore/usage#realtime-changes

Reply:
pourquoi faire ?

Reply:
Et useBlockForMaintenance => useCheckForMaintenance ?

Reply:
qu'est ce qui ne te plait pas dans useBlockForMaintenance ?

Reply:
on dirait que ça va bloquer l'app quoiqu'il arrive

Reply:
bon après c'est du wording c'est pas grave

Reply:
vu pour le unsubscribe, j'ai mis à jour